src/pycell.rs

```diff
@@ -263,6 +265,7 @@ unsafe impl<T, U> PyLayout<T> for PyCellBase<U> where U: PySizedLayout<T> {}
 /// ```
 /// For more information on how, when and why (not) to use `PyCell` please see the
 /// [module-level documentation](self).
+#[deprecated(since = "0.21.0")]
```

Comment:
I guess here we would want a deprecation message instructing users to use `Bound<T>` instead?

Shall we make this deprecation conditional on the `gil-refs` feature, given that we're very much treating `PyCell` as part of that API?

We might want to check the documentation and the guide to make sure it's clear that users can see that this concept of borrow tracking is not being removed from PyO3, just the `PyCell` type.

Reply:
I guess I just didn't completely finish that on 😅. I feature-gated it and added a small migration note.